src/raft.c

```diff
@@ -434,17 +471,27 @@ RedisModuleCallReply *RaftExecuteCommandArray(RedisRaftCtx *rr,
          * When we have an ACL, we will have a user set on the context, so need "C"
          */
         char *resp_call_fmt;
-        if (cmds->cmd_flags & CMD_SPEC_MULTI) {
+        if (cmds->cmd_flags & CMD_SPEC_MULTI || client_session) {
```

Comment:
Don't know what Redis currently does but can't we send WATCH and then BLPOP (without multi)? 
Btw, a test case might be good for this depending on the answer.

Reply:
needs to fix.